libguard/guard_interface.cpp

```diff
@@ -167,9 +171,20 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType)
             }
             return getHostEndiannessRecord(existGuard);
         }
+
+        // Storing the 1st resolved guard record position
```

Comment:
- I think this statement is not correct because we may have the record (which is matched with the given record id) that can be resolved and you are skipping that right?
- Do we need to use the record (which is matched with the given record id and that is resolved) slot to create a new record if no space in the GUARD file? What is the behavior in HostBoot for the same case?

Reply:
Why its wrong, I am storing the position not the record ID. I am not getting please elaborate. 
libguard is doing same as hostboot.

Reply:
Let's assume (we can store max 3 records) like below.

```
FFFFFFFF core0
00000001 core1
FFFFFFFF core2
```
Now trying to guard a `core0` then which slot you will consider?  1st slot or 3rd slot? If you think, 3rd then the statement  `Storing the 1st resolved guard record position` is correct?

(I think you won't get 1st slot position since we are trying the same entity (core0) which is already resolved so it moves to the next record based on above code logic) - correct me if I am wrong. 

If My guess is correct then, I like to know the behavior of the same case from the HostBoot side when you mentioned HB will use the first resolved record slot - IMO, we need to use the first slot (assume the above example) instead of 3rd slot since user/application trying to create a record for the entity which is already resolved if space is full.

Reply:
I meant 1st invalid/resolved record encountered so in the above case FFFFFFFF core0 will be used.
It doesnt matter if the entity path is same its the oldest invalid record present.

Reply:
> I meant 1st invalid/resolved record encountered so in the above case FFFFFFFF core0 will be used

How if trying to create `core0`?
https://github.com/chiragsibm/guard/commit/033591a341216f2af5f5994851a8f1156f036261#diff-afe02f82ae6baea2b83e14a0385bd9715fc2cd75ba556e86e68bd87984d40d09R163  - It won't go into the next record?

> It doesnt matter if the entity path is same its the oldest invalid record present.

I guess, it matters from the guard history use case if you are using the core2 instead of using the core1 slot.

Reply:
I meant oldest invalid/resolved record.
As per above example "FFFFFFFF core0" should be replaced.
Did the changes :)